test/test_functions.jl

```diff
@@ -150,10 +150,10 @@ function test_operator_implementation(
         end
 
         op=op_type(make_array(dim, ComplexF64, values))
-    
         Snowflake.apply_operator!(ψ_1,op,[v for v in 1:get_num_qubits(op)])
-
-        @test op*ψ_0 ≈ ψ_1
+        if (op_type!=SparseOperator)
```

Comment:
This doesn't seem right. This is making the functions do different things for different operators (more than one thing) Either make the function generic enough so that this escape hatch is not necessary or split the function in two.